### PR TITLE
refactor(server): move asset detail endpoint to new controller

### PR DESCRIPTION
**Pull request by jrasm91**
> [!NOTE]
> `/asset/assetById/:id` is deprecated and will be removed in a future release

- Rename api-v1/asset/asset-repository to `*V1` everywhere
- Add a new `/asset/:id` endpoint (in the new controller)
- Move the existing service method to the new service
- Point the old method at the new service
- Delete the old repository implementation and use the new one instead
- Added e2e tests for the asset detail endpoint

Note: A lot of relations where loaded for `serveFile`, but now it should just load the asset, so that endpoint should be faster.

### CODE REVIEW

File: web/src/lib/components/asset-viewer/detail-panel.svelte [hunk 1]
```diff
@@ -234,7 +234,7 @@
       {/key}
     </section>
   {:else if description}
-    <p class="break-words whitespace-pre-line w-full text-black dark:text-white text-base">{description}</p>
+    <p class="px-4 break-words whitespace-pre-line w-full text-black dark:text-white text-base">{description}</p>
```
Comment (jrasm91) [inline web/src/lib/components/asset-viewer/detail-panel.svelte:237]: This seemed like a regression that I fixed here while testing the asset detail endpoint. cc: @martabal

File: server/src/domain/asset/asset.service.ts [hunk 2]
```diff
@@ -400,6 +398,44 @@ export class AssetService {
...
+      }
+
+      if (auth.sharedLink) {
+        delete data.owner;
+      }
+
+      return data;
+    } else {
+      return mapAsset(asset, { stripMetadata: true, withStack: true });
+    }
```
Comment (danieldietzler) [inline server/src/domain/asset/asset.service.ts:436]: Can we flip this? So like `if (auth.SharedLink && !auth.sharedLink?.showExif) {return mapAsset(asset, { stripMetadata: true, withStack: true });}` and then all the other stuff below without an else. I prefer having as little deep code as possible

**Review by danieldietzler — APPROVED** [review]
Nice! Making progress in getting rid of api-v1 is always great :)

File: server/src/domain/asset/asset.service.ts (hunk 2 above)
Comment (jrasm91) [reply server/src/domain/asset/asset.service.ts:436]: Yeah, that makes sense. I copied it as is from the other one, but I like that approach better.

**Pull request merged**